src/test/java/ibm/jceplus/junit/base/BaseTestAESGCMCipherInputStreamExceptions.java

```diff
@@ -343,33 +226,31 @@ private void do_cbc_readAllIllegalBlockSize() throws Exception {
             while ((s = in.read(read)) != -1) {
                 size += s;
             }
-            throw new RuntimeException("Fail: No IllegalBlockSizeException. "
-                    + "CipherInputStream.read() returned " + size);
-
-        } catch (IOException e) {
-            Throwable ec = e.getCause();
-            if (ec instanceof IllegalBlockSizeException) {
-                System.out.println("  Pass.");
-            } else {
-                System.out.println("  Fail: " + ec.getMessage());
-                throw new RuntimeException(ec);
+            fail("IOException expected.");
+        } catch (Exception e) {
```

Comment:
Yes changed this to look for IOException. The test should fail otherwise. 

We are looking for a IOException here as this is what the code currently throws. ( This IOException just happens to contain a `IllegalBlockSizeException` for the record )